pkg/splunk/enterprise/util.go

```diff
@@ -292,3 +294,236 @@ func DeleteOwnerReferencesForS3SecretObjects(client splcommon.ControllerClient,
+	appDeployContext *enterprisev1.AppDeploymentContext, remoteObjListingMap map[string]splclient.S3Response, appFrameworkConfig *enterprisev1.AppFrameworkSpec) (bool, error) {
+	crKind := cr.GetObjectKind().GroupVersionKind().Kind
+	scopedLog := log.WithName("handleAppRepoChanges").WithValues("kind", crKind, "name", cr.GetName(), "namespace", cr.GetNamespace())
+	var err error
+
+	scopedLog.Info("App Listing", " received App listing for %d app sources", len(remoteObjListingMap))
+	if remoteObjListingMap == nil || len(remoteObjListingMap) == 0 {
+		scopedLog.Error(nil, "remoteObjectList is empty")
+	}
+
+	// Check if the appSource is  still valid in the config
+	for appSrc := range remoteObjListingMap {
+		if !CheckIfAppSrcExistsInConfig(appFrameworkConfig, appSrc) {
+			err = fmt.Errorf("App source: %s is no more exists, this should never happen", appSrc)
+			return false, err
+		}
+	}
+
+	// ToDo: sgontla: Ideally, this check should go to the reconcile entry point once the glue logic in place.
+	if appDeployContext.AppsSrcDeployStatus == nil {
+		appDeployContext.AppsSrcDeployStatus = make(map[string]enterprisev1.AppSrcDeployInfo)
+	}
+
+	// 1. Go through each AppSrc in my local list, and look for any changes
+	for appSrc, appSrcDeploymentInfo := range appDeployContext.AppsSrcDeployStatus {
+		// 1.1 Check if the AppSrc is deleted in latest config, is so, mark them for deletion
+		if !CheckIfAppSrcExistsInConfig(appFrameworkConfig, appSrc) ||
+			!checkIfAppSrcExistsWithRemoteListing(appSrc, remoteObjListingMap) {
+			scopedLog.Info("App Listing:", " deleting/disabling all the apps for App source: %s. Reason: App source deleted", appSrc)
+			curAppDeployList := appSrcDeploymentInfo.AppDeploymentInfoList
+			if setStateAndStatusForAppDeployInfoList(curAppDeployList, enterprisev1.RepoStateDeleted, enterprisev1.DeployStatusPending) {
+				appDeployContext.IsDeploymentInProgress = true
+			}
+		}
+
+		// Finally update the Map entry with latest info
+		appDeployContext.AppsSrcDeployStatus[appSrc] = appSrcDeploymentInfo
+	}
+
+	// 2. Go through each AppSrc from the remote listing
+	for appSrc, s3Response := range remoteObjListingMap {
+		// 2.1 Mark Apps for deletion if they are missing in remote listing
+		appSrcDeploymentInfo, appSrcExistsLocally := appDeployContext.AppsSrcDeployStatus[appSrc]
+
+		if appSrcExistsLocally {
+			currentList := appSrcDeploymentInfo.AppDeploymentInfoList
+			for appIdx := range currentList {
+				if !checkIfAnAppIsActiveOnRemoteStore(currentList[appIdx].AppName, s3Response.Objects) {
+					setStateAndStatusForAppDeployInfo(currentList[appIdx], enterprisev1.RepoStateDeleted, enterprisev1.DeployStatusPending)
+					appDeployContext.IsDeploymentInProgress = true
+				}
+			}
+		}
+
+		// 2.2 Check for any App changes(Ex. A new App source, a new App added/updated)
+		if createOrUpdateAppSrcDeploymentInfo(&appSrcDeploymentInfo, s3Response.Objects) {
+			appDeployContext.IsDeploymentInProgress = true
```

Comment:
So is it correct to say that if the `IsDeploymentInProgress` flag is true, then theres a change in the Apps we need to deployment and we can use it as a trigger to kick off App installation (via Ansible/pod restart)?